src/main/kotlin/com/sourcegraph/cody/chat/PromptHistory.kt

```diff
@@ -0,0 +1,34 @@
+package com.sourcegraph.cody.chat
+
+open class PromptHistory(private val capacity: Int) {
+  private val history = mutableListOf<String>()
+  private var currentIndex = 0
+
+  fun add(item: String) {
+    history.add(item)
+    if (history.size > capacity) {
+      history.removeAt(0)
+    }
+    resetHistory()
+  }
+
+  fun getPrevious(): String? {
+    if (currentIndex > 0) {
+      currentIndex--
+    }
+    return history.getOrNull(currentIndex)
+  }
+
+  fun getNext(): String? {
+    if (currentIndex < history.size) {
```

Comment:
Shouldn't it be `if (currentIndex < history.size - 1)`?

Reply:
No, we want to get `null` at the end here. Based on that we determine that we want to show a clear input field again.

e.g. with the history:
history[0] = "hi"
history[1] = "sup?"
and the text area:
textArea.text = ""

1. ⬆️ 
2. textArea.text = "sup?"
3. ⬇️ 
4. textArea.text = ""

Reply:
Ok, cool. It's a bit misleading that `getPrevious` have the same signature and code layout, but in it's case we prevent the index to go bellow 0, but I understand the intention is that `null` can happen there if `history` is empty?

Reply:
Yes, it is possible but handled by the history manager and covered by tests (unit tests and my manual tests)